packages/nextjs/app/builders/0x972d1be405DcFcD32DD95743934693B673e3F2B0/page.tsx

```diff
@@ -0,0 +1,126 @@
+import Image from "next/image";
+import { NextPage } from "next";
+import { IconType } from "react-icons";
+import { FaEnvelope, FaGithub, FaMapMarkerAlt } from "react-icons/fa";
+
+interface ProfileData {
+  name: string;
+  title: string;
+  bio: string;
+  avatar: string;
+  location: string;
+  email: string;
+  social: {
+    github: string;
+    twitter: string;
+  };
+}
+
+interface StyledIconProps extends React.HTMLAttributes<HTMLSpanElement> {
+  icon: IconType;
+  size?: number;
+  color?: string;
+}
+
+interface SocialLink {
+  href: string;
+  icon: IconType;
+  lightHoverColor: string;
+  darkHoverColor: string;
+}
+
+const StyledIcon: React.FC<StyledIconProps> = ({ icon: Icon, className, size, color, ...props }) => {
+  return (
+    <span className={`inline-block ${className}`} {...props}>
+      <Icon size={size} color={color} />
+    </span>
+  );
+};
```

Comment:
By creating this component, you’re abstracting very little code. And if one of the icons needs a different customization in the future, it will be harder to handle. So this code is actually harder to maintain than if you had kept it simple.